### DISCUSSION TITLE
Improvement for formatting by webstorm plugin

### DISCUSSION THREAD

**Discussion by Arctomachine** [biomejs/biome#554] [RFC]
Right now plugin interferes with default auto format in projects where biome is not used at all.

> java.util.concurrent.ExecutionException: com.intellij.execution.ExecutionException: Biome language server is not found, make sure you have @biomejs/biome installed.

Instead I suggest following decision flow to make it more flexible
![Untitled](https://github.com/biomejs/biome/assets/29041820/35e1c8bb-2cf8-493c-9935-df8ea9eca7b4)

**Comment by victor-teles** [comment]
@Arctomachine 

## Plugin activation flow
Your flow idea is great!

I'd also like to suggest a new configuration option for checking biome.json or local dependency, because the `biome` could also be used globally.

So maybe we can introduce a new option (enabled by default):

- Require configuration or local dependency (like in biome's vscode extension)

## Formatter

[Jetbrains recently released support for native formatting using LSP](https://discord.com/channels/1132231889290285117/1132614451565170819/1160866398147452999)

With this I believe we can achieve something more stable in terms of formatting, but we need to try out.
